proxy_agent/src/provision.rs

```diff
@@ -321,49 +319,34 @@ pub async fn get_provision_status_wait(port: u16, duration: Option<Duration>) ->
 // return value
 //  bool - true provision finished; false provision not finished
 //  String - provision error message, empty means provision success or provision failed.
-fn get_current_provision_status(port: u16) -> std::io::Result<ProivsionState> {
-    let provision_url = url::Url::parse(&format!(
+async fn get_current_provision_status(port: u16) -> std::io::Result<ProivsionState> {
+    let provision_url: hyper::Uri = format!(
         "http://{}:{}{}",
         Ipv4Addr::LOCALHOST,
         port,
         PROVISION_URL_PATH
-    ))
+    )
+    .parse()
     .map_err(|e| {
         std::io::Error::new(
             std::io::ErrorKind::Other,
             format!("Failed to parse provision url with error: {}", e),
         )
     })?;
-    let mut req = Request::new(PROVISION_URL_PATH.to_string(), "GET".to_string());
-    req.headers
-        .add_header(constants::METADATA_HEADER.to_string(), "True ".to_string());
-
-    let mut http_request = HttpRequest::new(provision_url, req);
-    http_request
-        .request
-        .headers
-        .add_header("Host".to_string(), http_request.get_host());
-
-    let response = http::get_response_in_string(&mut http_request)?;
-    let response_body = response.get_body_as_string()?;
-    if response.status != Response::OK {
-        return Err(Error::new(
-            ErrorKind::Other,
-            format!("Host response {} - {}", response.status, response_body),
-        ));
-    }
 
-    let state: ProivsionState = serde_json::from_str(&response_body)?;
-    Ok(state)
+    let mut headers = HashMap::new();
+    headers.insert(constants::METADATA_HEADER.to_string(), "true".to_string());
```

Comment:
does it miss a 'Host" header in the http request?

Reply:
host header is common one, it will be added always in fn build_request.